### PR TITLE
ask about bicycle shop details

### LINKED ISSUE #4012: Ask for services offered by a bicycle shop
### General
Affected tag(s) to be modified/added: [service:bicycle:retail](https://wiki.openstreetmap.org/wiki/Tag:shop%3Dbicycle)  [service:bicycle:repair](https://wiki.openstreetmap.org/wiki/Tag:shop%3Dbicycle) [service:bicycle:rental](https://wiki.openstreetmap.org/wiki/Tag:shop%3Dbicycle) ...
Question asked: **Mark services offered by this bicycle shop** / **Which services are offered by this bicycle shop?**

### Checklist
Checklist for quest suggestions (see [guidelines](https://github.com/streetcomplete/StreetComplete/blob/master/QUEST_GUIDELINES.md)):
- [x] 🚧 To be added tag is established and has a useful purpose
- [x] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one)
- [ ] 🐿️ Easily answerable by everyone from the outside but a survey is necessary - may be necessary to disable it by default
- [x] 💤 Not an overwhelming percentage of quests have the same answer (No spam)
- [x] 🕓 Applies to a reasonable number of map data (Worth the effort)

### Ideas for implementation

**Element selection:** 
Skip objects where all (or any, or most) of this objects is tagged and all objects where any of this tags is present and was tagged recently
 

**Metadata needed:** 
None

**Proposed UI:**
Maybe resurvey with prefilled currently available things is a good idea?

### PR DESCRIPTION
**Pull request by matkoniecz**
fixes #4012

note: icons are missing

@westnordost Are you interested in making something for iicons here? Or should I try?

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -668,6 +668,14 @@ However, before uploading your changes, the app checks with a &lt;a href=\"https
     <string name="quest_bicycle_rental_type_human">Human-operated bike rental</string>
     <string name="quest_bicycle_rental_type_shop_with_rental">Bike store that also rents bikes</string>
 
+    <string name="quest_bicycle_shop_repair_title">Can you repair bicycles here?</string>
+
+    <string name="quest_bicycle_shop_second_hand_title">What type of bicycles can be bought here?</string>
+    <string name="quest_bicycle_shop_second_hand_only_new">Only new bicycles</string>
+    <string name="quest_bicycle_shop_second_hand_new_and_used">New bicycles and second-hand bicycles</string>
+    <string name="quest_bicycle_shop_second_hand_only_used">Only second-hand bicycles</string>
+    <string name="quest_bicycle_shop_second_hand_no_bicycles">No bicycles sold in this bicycle shop</string>
```
Comment (FloEdelmann) [inline app/src/main/res/values/strings.xml:677]:
```suggestion
    <string name="quest_bicycle_shop_second_hand_title">Are second-hand bicycles sold here?</string>
    <string name="quest_bicycle_shop_second_hand_only_new">No, only new ones</string>
    <string name="quest_bicycle_shop_second_hand_new_and_used">Yes, alongside new ones</string>
    <string name="quest_bicycle_shop_second_hand_only_used">Yes, exclusively</string>
    <string name="quest_bicycle_shop_second_hand_no_bicycles">No bicycles sold at all</string>
```

File: app/src/main/java/de/westnordost/streetcomplete/quests/bike_shop/AddSecondHandBicycleAvailability.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            }
+            SecondHandBicycleAvailability.ONLY_SECOND_HAND -> {
+                tags.updateWithCheckDate("service:bicycle:retail", "yes")
+                tags["service:bicycle:second_hand"] = "only"
+            }
+            SecondHandBicycleAvailability.NO_BICYCLES_SOLD -> {
+                tags.updateWithCheckDate("service:bicycle:retail", "no")
+            }
+        }
+    }
```
Comment (FloEdelmann) [inline app/src/main/java/de/westnordost/streetcomplete/quests/bike_shop/AddSecondHandBicycleAvailability.kt:59]: Maybe extract the OSM tag values into variables of the `SecondHandBicycleAvailability` enum?

```kt
enum class SecondHandBicycleAvailability(osmValue: String?, retail: Boolean) {
    ONLY_NEW("no", true),
    NEW_AND_SECOND_HAND("yes", true),
    ONLY_SECOND_HAND("only", true),
    NO_BICYCLES_SOLD(null, false),
}
```

```suggestion
    override fun applyAnswerTo(answer: SecondHandBicycleAvailability, tags: Tags, timestampEdited: Long) {
        tags.updateWithCheckDate("service:bicycle:retail", answer.retail.toYesNo())
        answer.osmValue?.let { tags["service:bicycle:second_hand"] = it }
    }
```

File: app/src/main/java/de/westnordost/streetcomplete/quests/bike_shop/AddSecondHandBicycleAvailability.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        )
+        and (
+            service:bicycle:retail != no
+            or service:bicycle:retail older today -6 years
+        )
+        and !second_hand
+        and access !~ private|no
+        """
+
+    override val changesetComment = "Add type of parking access"
```
Comment (FloEdelmann) [inline app/src/main/java/de/westnordost/streetcomplete/quests/bike_shop/AddSecondHandBicycleAvailability.kt:28]:
```suggestion
    override val changesetComment = "Add whether bicycle shop sells second-hand bicycles"
```

File: app/src/main/java/de/westnordost/streetcomplete/quests/bike_shop/AddSecondHandBicycleAvailability.kt (hunk 2 above)
Comment (FloEdelmann) [reply app/src/main/java/de/westnordost/streetcomplete/quests/bike_shop/AddSecondHandBicycleAvailability.kt:59]: Or maybe with only one variable:

```kt
enum class SecondHandBicycleAvailability(osmValue: String?) {
    ONLY_NEW("no"),
    NEW_AND_SECOND_HAND("yes"),
    ONLY_SECOND_HAND("only"),
    NO_BICYCLES_SOLD(null),
}
```

```suggestion
    override fun applyAnswerTo(answer: SecondHandBicycleAvailability, tags: Tags, timestampEdited: Long) {
        if (answer.osmValue == null) {
            tags.updateWithCheckDate("service:bicycle:retail", "no")
        } else {
            tags.updateWithCheckDate("service:bicycle:retail", "yes")
            tags["service:bicycle:second_hand"] = answer.osmValue
        }
    }
```

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (matkoniecz) [reply app/src/main/res/values/strings.xml:677]: I had in one of drafts and I was wondering is it a bit too tricky. I guess I will go with it.

I just want to defend `No bicycles sold in this bicycle shop` vs `No bicycles sold at all` - if bicycle shop is replaced by say supermarket of greengrocer, them `No bicycles sold at all` is still applicable - but user should leave note instead.

Comment (FloEdelmann) [reply app/src/main/res/values/strings.xml:677]: Shouldn't the feature name "Bicycle shop" below the quest title suffice to raise concerns if there is e.g. a supermarket there now?

But I guess that being explicit doesn't hurt. It just sounds a bit repetitive.

Comment (matkoniecz) [reply app/src/main/res/values/strings.xml:677]: > Shouldn't the feature name "Bicycle shop" below the quest title suffice to raise concerns if there is e.g. a supermarket there now?

It should but I estimate that at least 10%, maybe up to 60% of people will miss it.

**Review by westnordost — COMMENTED** [review]
Why is "second hand bikes" quest disabled by default but "bike repair services" not?

Yes, I can do icons

**Comment by matkoniecz** [comment]
> Why is "second hand bikes" quest disabled by default but "bike repair services" not?

I expect one to require checking inside and one not, but if that is based too much on what I personally encountered I can apply the same "disabled by default" to both

**Comment by westnordost** [comment]
> I expect one to require checking inside and one not, but if that is based too much on what I personally encountered I can apply the same "disabled by default" to both

I don't know, if I see a couple of obbviously used bikes in front of a bike shop, it could be **either** these are to be repaired / repaired bikes or it could be that they are selling second hand bikes.

**Comment by westnordost** [comment]
Added icons

File: app/src/main/java/de/westnordost/streetcomplete/quests/bike_shop/AddBikeRepairAvailability.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            !service:bicycle:repair
+            or service:bicycle:repair older today -6 years
+        )
+        and access !~ private|no
+    """
+
+    override val changesetComment = "Add whether bicycle shop offers repair"
+    override val defaultDisabledMessage = R.string.default_disabled_msg_go_inside
+    override val wikiLink = "Key:service:bicycle:repair"
+    override val icon = R.drawable.ic_quest_car_air_compressor // TODO ADD AN ICON
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/quests/bike_shop/AddBikeRepairAvailability.kt:29]:
```suggestion
    override val icon = R.drawable.ic_quest_bicycle_repair
```

**Review by westnordost — APPROVED** [review]

**Pull request merged**